### ISSUE TITLE
gcc 5.1.0 --with-java fails to build on 10.11

### ISSUE THREAD

**Issue by aw1621107** [Homebrew/homebrew#41633]
[gist](https://gist.github.com/anonymous/80cdf9a02894ddbd08ea)

Clean install of 10.11, XCode 7.0. Compile seems to make it past the stage 2/3 comparison, fails somewhere when dealing with gfortran? Compiles and installs fine when passed --without-fortran, ~~even with --with-java/git/nls. --with-all-languages seems to override --without-fortran, though?~~

There seems to be some errors when compiling nls stuff, around line ~37000 in 02.make, but it doesn't seem to stop the build process.

If any more info is needed, let me know. My apologies for not being able to provide more useful info; most of this is gibberish to me.

**Comment by tdsmith** [comment]
I can successfully build gcc on this configuration on 10.11 PB1:
```
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew
HEAD: 3947f85477e93e0bb0a6cc880b948204102c195c
Last commit: 7 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: dual-core 64-bit haswell
OS X: 10.11-x86_64
Xcode: N/A
CLT: 7.0.0.0.1.1436025195
Clang: 7.0 build 700
X11: N/A
System Ruby: 2.0.0-p645
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby
Java: N/A
```

**Comment by tdsmith** [comment]
I think this is the error that killed this build:
```
depbase=`echo java/lang/reflect/natVMProxy.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
        /bin/sh ./libtool --tag=CXX   --mode=compile /private/tmp/gcc20150712-81848-xw8vaz/gcc-5.1.0/build/./gcc/xgcc -shared
-libgcc -B/private/tmp/gcc20150712-81848-xw8vaz/gcc-5.1.0/build/./gcc -nostdinc++ -L/private/tmp/gcc20150712-81848-xw8vaz/gcc
-5.1.0/build/x86_64-apple-darwin15.0.0/i386/libstdc++-v3/src -L/private/tmp/gcc20150712-81848-xw8vaz/gcc-5.1.0/build/x86_64-a
pple-darwin15.0.0/i386/libstdc++-v3/src/.libs -L/private/tmp/gcc20150712-81848-xw8vaz/gcc-5.1.0/build/x86_64-apple-darwin15.0
.0/i386/libstdc++-v3/libsupc++/.libs -B/usr/local/Cellar/gcc/5.1.0/x86_64-apple-darwin15.0.0/bin/ -B/usr/local/Cellar/gcc/5.1
.0/x86_64-apple-darwin15.0.0/lib/ -isystem /usr/local/Cellar/gcc/5.1.0/x86_64-apple-darwin15.0.0/include -isystem /usr/local/
Cellar/gcc/5.1.0/x86_64-apple-darwin15.0.0/sys-include  -m32 -DHAVE_CONFIG_H -I. -I../../../../libjava -I./include -I./gcj  -
I../../../../libjava -Iinclude -I../../../../libjava/include -I../../../../libjava/classpath/include -Iclasspath/include -I..
/../../../libjava/classpath/native/fdlibm -I../../../../libjava/../boehm-gc/include -I../boehm-gc/include  -I../../../../libj
ava/libltdl -I../../../../libjava/libltdl -I../../../../libjava/.././libjava/../libgcc  -I../../../../libjava/../libffi/inclu
de -I../libffi/include  -fno-rtti -fnon-call-exceptions  -fdollars-in-identifiers -Wswitch-enum -D_FILE_OFFSET_BITS=64 -ffloa
t-store -fomit-frame-pointer -Usun -Wextra -Wall -D_GNU_SOURCE -DPREFIX="\"/usr/local/Cellar/gcc/5.1.0\"" -DTOOLEXECLIBDIR="\"/usr/local/Cellar/gcc/5.1.0/lib/gcc/5/i386\"" -DJAVA_HOME="\"/usr/local/Cellar/gcc/5.1.0\"" -DBOOT_CLASS_PATH="\"/usr/local/Cellar/gcc/5.1.0/share/java/libgcj-5.1.0.jar\"" -DJAVA_EXT_DIRS="\"/usr/local/Cellar/gcc/5.1.0/share/java/ext\"" -DGCJ_ENDORSED_DIRS="\"/usr/local/Cellar/gcc/5.1.0/share/java/gcj-endorsed\"" -DGCJ_VERSIONED_LIBDIR="\"/usr/local/Cellar/gcc/5.1.0/lib/gcc/5/i386/gcj-5.1.0-16\"" -DPATH_SEPARATOR="\":\"" -DECJ_JAR_FILE="\"/usr/local/opt/ecj/share/java/ecj.jar\"" -DLIBGCJ_DEFAULT_DATABASE="\"/usr/local/Cellar/gcc/5.1.0/lib/gcc/5/i386/gcj-5.1.0-16/classmap.db\"" -DLIBGCJ_DEFAULT_DATABASE_PATH_TAIL="\"gcj-5.1.0-16/classmap.db\"" -g -O2  -m32 -MT java/lang/reflect/natVMProxy.lo -MD -MP -MF $depbase.Tpo -c -o java/lang/reflect/natVMProxy.lo ../../../../libjava/java/lang/reflect/natVMProxy.cc &&\
        mv -f $depbase.Tpo $depbase.Plo
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:1110:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    8(%esi,%edi,2)
        ^
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:1120:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    10(%esp)
        ^
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:1267:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    8(%esi,%edi,2)
        ^
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:1277:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    14(%esp)
        ^
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:1830:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    14(%esp)
        ^
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:1907:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    14(%esp)
        ^
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:1988:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    14(%esp)
        ^
/var/folders/yg/_l5j_k5x0sn2m138tyrc0ylc0000gn/T//ccsxlyJd.s:2047:2: error: ambiguous instructions require an explicit suffix (could be 'filds', or 'fildl')
        fild    14(%esp)
        ^
libtool: compile:  /private/tmp/gcc20150712-81848-xw8vaz/gcc-5.1.0/build/./gcc/xgcc -B/private/tmp/gcc20150712-81848-xw8vaz/gcc-5.1.0/build/./gcc/ -B/usr/local/Cellar/gcc/5.1.0/x86_64-apple-darwin15.0.0/bin/ -B/usr/local/Cellar/gcc/5.1.0/x86_64-apple-darwin15.0.0/lib/ -isystem /usr/local/Cellar/gcc/5.1.0/x86_64-apple-darwin15.0.0/include -isystem /usr/local/Cellar/gcc/5.1.0/x86_64-apple-darwin15.0.0/sys-include -DHAVE_CONFIG_H -I. -I../../../libgfortran -iquote../../../libgfortran/io -I../../../libgfortran/../gcc -I../../../libgfortran/../gcc/config -I../../../libgfortran/../libquadmath -I../.././gcc -I../../../libgfortran/../libgcc -I../libgcc -std=gnu11 -Wall -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wextra -Wwrite-strings -Werror=implicit-function-declaration -Werror=vla -fcx-fortran-rules -ffunction-sections -fdata-sections -g -O2 -MT product_i1.lo -MD -MP -MF .deps/product_i1.Tpo -c ../../../libgfortran/generated/product_i1.c -o product_i1.o >/dev/null 2>&1
make[5]: *** [java/lang/reflect/natArray.lo] Error 1
make[5]: *** Waiting for unfinished jobs....
```

**Comment by tdsmith** [comment]
note that the proximate failure here is Java-related, not fortran-related, despite the libtool line at the bottom, which I expect is just the result of parallel processes printing at the same time.

**Comment by tdsmith** [comment]
`brew install gcc --with-java` fails for me the same way it does for you -- note that the output is cleaner on two cores: https://gist.github.com/9421f3e8fd864340dfbf

**Comment by tdsmith** [comment]
`brew install gcc --with-java --without-fortran` fails: https://gist.github.com/0f30cacee67e1d5a35ca

**Comment by aw1621107** [comment]
Whoops, should have been more thorough with the flags. Got the same results as you.

**Comment by tdsmith** [comment]
Upstream bug, looks like a patch is available: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66509

**Comment by tdsmith** [comment]
Should be fixed by ebd659a; thanks for the report!
